### DISCUSSION TITLE
Javaagent jar without version name in the file name

### DISCUSSION THREAD

**Discussion by itsmaneesh** [open-telemetry/opentelemetry-java-instrumentation#9266] [Q&A]
Hi,
Is there a way to get javaagent jar without a version number associated with it.?
For e.g.: Get opentelemetry-javaagent.jar instead as opentelemetry-javaagent-1.26.0.jar 

So that, we avoid changes to our code whenever there is a new one version of javaagent is released?

Rationale for this use case:
I have a lib called `my-lib` that gets the opentelemetry-javaagent as gradle `runtimeOnly` dependency.
Also have a service `my-service` that use helm chart has been updated to have the below configuration:
- Use `my-lib` as `implementation` dependency so when I deploy the `my-service` service then `opentelemetry-javaagent-1.26.0.jar` gets added to the `/app/libs` directory 
- Use generic helm chart `generic-chart` (used by many services in our org) that configures the javaagent path using JAVA_TOOL_OPTIONS env variable and other env variables include OTEL specfic variables as well.
          fpr e.g.: JAVA_TOOL_OPTIONS=-javaagent:/app/libs/opentelemetry-javaagent-1.26.0.jar 

In this above scenario, the javaagent version is tightly coupled between the `my-lib` release version and `generic-chart` release version. 
If I want to use the latest version of opentelemetry javaagent then I have to upgrade both `my-lib` to use the latest version of opentelemetry javaagent in dependencies and also update the `generic-chart` to use the latest version jar name in the JAVA_TOOL_OPTIONS and then release a breaking change for both of them.
i.e: If any service use the latest version of `my-lib` (that has 1.26.0 javaagent) then they HAVE to use latest version of `generic-chart` that adds opentelemetry-javaagent-1.26.0.jar as  JAVA_TOOL_OPTIONS, otherwise if service will not start due to javaagent jar file  missing.

To avoid such breaking changes, I would like to have `my-lib` to get the opentelemetry-javaagent.jar (without version number in thr file name) and use JAVA_TOOL_OPTIONS=-javaagent:/app/libs/opentelemetry-javaagent.jar 
So, can you please advice on getting javaagent jar without version number in the file name?

Thanks.

**Comment by trask** [comment]
Does https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v1.29.0/opentelemetry-javaagent.jar work for you?

**Reply by itsmaneesh** [reply]
@trask Thank you for the above option of downloading the jar from the releases page. But, I am looking for an option of having this jar (without version number in the file name) as gradle dependency in my lib. ? Is there a way to do so?

**Reply by trask** [reply]
does this help? https://stackoverflow.com/questions/10370422/gradle-getting-the-latest-release-version-of-a-dependency
